yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -473,7 +477,50 @@ func (p *ParserIssueDetector) genericIssues(query string) ([]QueryIssue, error)
 	return result, nil
 }
 
-func (p *ParserIssueDetector) GetRedundantIndexIssues(redundantIndexes []utils.RedundantIndexesInfo) []QueryIssue {
+func (p *ParserIssueDetector) getJsonbReturnTypeFunctions() []string {
+	var jsonbFunctions []string
+	jsonbColumns := p.jsonbColumns
+	for _, function := range p.functionObjects {
+		returnType := function.ReturnType
+		if strings.HasSuffix(returnType, "%TYPE") {
+			// e.g. public.table_name.column%TYPE
+			qualifiedColumn := strings.TrimSuffix(returnType, "%TYPE")
+			parts := strings.Split(qualifiedColumn, ".")
+			column := parts[len(parts)-1]
+			if slices.Contains(jsonbColumns, column) {
+				jsonbFunctions = append(jsonbFunctions, function.FuncName)
+			}
+		} else {
+			// e.g. public.udt_type, text, trigger, jsonb
+			parts := strings.Split(returnType, ".")
+			typeName := parts[len(parts)-1]
+			if typeName == "jsonb" {
+				jsonbFunctions = append(jsonbFunctions, function.FuncName)
+			}
+		}
+	}
+	jsonbFunctions = append(jsonbFunctions, catalogFunctionsReturningJsonb.ToSlice()...)
+	return jsonbFunctions
+}
+
+func (p *ParserIssueDetector) IsGinIndexPresentInSchema() bool {
+	return p.isGinIndexPresentInSchema
+}
+
+func (p *ParserIssueDetector) IsUnloggedTablesIssueFiltered() bool {
+	return p.isUnloggedTablesIssueFiltered
+}
+
+func (p *ParserIssueDetector) PopulateColumnStatisticsMap(columnStats []utils.ColumnStatistics) {
+	for _, stat := range columnStats {
+		qualifiedColumnName := fmt.Sprintf("%s.%s", stat.GetTableName(), stat.ColumnName)
+		p.columnStatistics[qualifiedColumnName] = stat
+	}
+}
+
+// ======= Functions not use parser right now
+
+func GetRedundantIndexIssues(redundantIndexes []utils.RedundantIndexesInfo) []QueryIssue {
```

Comment:
hmm, i guess eventually in the long run we will move this as well into the parserIssueDetector. 
Makes sense for now, for this to be separate 👍